src/main/java/com/shapesecurity/shift/es2016/parser/GenericParser.java

```diff
@@ -1920,14 +1929,15 @@ protected BindingBindingWithDefault targetToBindingPossiblyWithDefault(@NotNull
             }
         }
 
+        FormalParameters paramsNode = this.finishArrowParams(params, Maybe.empty(), postParenStartState);
```

Comment:
I'm not super thrilled with this - we end up creating this node even though we might not use it. I don't see a way around it given the current API, though.